keystorage/aws/src/main/java/tech/pegasys/signers/aws/SecretValueConverterUtil.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.aws;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class SecretValueConverterUtil {
+  private static final Logger LOG = LogManager.getLogger();
+
+  public static <R> Set<R> mapSecretValue(
+      BiFunction<String, String, R> mapper, String propKey, String secretValue) {
+    final AtomicLong discardedValues = new AtomicLong(0);
+    final Set<R> mappedValues =
+        secretValue
+            .lines()
+            .map(
+                v -> {
+                  final R obj = mapper.apply(propKey, v);
+                  if (obj == null) {
+                    discardedValues.incrementAndGet();
+                  }
+                  return obj;
+                })
+            .filter(Objects::nonNull)
+            .collect(Collectors.toSet());
+
+    if (discardedValues.get() > 0) {
+      LOG.warn(
+          "AWS Secrets - {} values from property {} were not mapped and discarded.",
```

Comment:
Is there value in knowing which values were discarded?

Reply:
no .... since they can technically be private keys (or some combination of) ... don't really want to log the values.

Reply:
Definitely should not log values.

Knowing which index out of a large list of values would be helpful though to help pinpoint the bad key though.

Reply:
Key + index then?